### PR TITLE
Added color removal and fixed duplicate observation bug for templates

### PR DESCRIPTION
**Pull request by samhum**
Changed DB schema for new column for color stripped observation. Fixed bug where templated observations show up twice in db. Before adding new the column, the bukkit strip color method worked perfectly but after left &s and an extra chars so made a little algorithm to fix it (could be cleaner) but will mess up observations if user uses &.

### CODE REVIEW

File: src/main/java/edu/whimc/observations/observetemplate/gui/TemplateSelection.java
Comment (jackah2) [inline src/main/java/edu/whimc/observations/observetemplate/gui/TemplateSelection.java:206]: I'd prefer this logic only be implemented once in a static helper function inside `Observation.java`. [This](https://github.com/whimc/Observations/blob/0805ac76a2ae7db874737d74f582b913aaba2109/src/main/java/edu/whimc/observations/commands/ObserveCommand.java#L34-L46) should be moved to `Observation.java` and refactored to accept a template for the observation. The method name should be changed to be indicative that it uses the current time and perhaps also that it triggers the Bukkit event.

Comment (jackah2) [reply src/main/java/edu/whimc/observations/observetemplate/gui/TemplateSelection.java:206]: Also, this is where the duplicated logic is coming from. The observation creation is implicitly handled in the call of `sendFooter` since the `withConfirm` parameter is `true`. [Lines 202-210](https://github.com/whimc/Observations/blob/7b47fcca90e57b6eff9ee80c08e994b38c7f5c8b/src/main/java/edu/whimc/observations/observetemplate/gui/TemplateSelection.java#L202-L210) can be removed while [lines 253-257](https://github.com/whimc/Observations/blob/7b47fcca90e57b6eff9ee80c08e994b38c7f5c8b/src/main/java/edu/whimc/observations/observetemplate/gui/TemplateSelection.java#L253-L257) should be refactored to use the new helper method mentioned above.

File: src/main/java/edu/whimc/observations/utils/sql/Queryer.java
Comment (jackah2) [inline src/main/java/edu/whimc/observations/utils/sql/Queryer.java:143]:
```suggestion
        statement.setString(14, ChatColor.stripColor(Utils.color(obs.getObservation())));
```
A cleaner way to do this would be to color the string then strip the colors so you don't have to manually do any string parsing.

File: src/main/java/edu/whimc/observations/utils/sql/migration/schemas/Schema_4.java
Comment (jackah2) [inline src/main/java/edu/whimc/observations/utils/sql/migration/schemas/Schema_4.java:25]: File should have extra newline

File: src/main/java/edu/whimc/observations/utils/sql/migration/schemas/Schema_4.java
Comment (jackah2) [inline src/main/java/edu/whimc/observations/utils/sql/migration/schemas/Schema_4.java:21]: Would be good to also modify the existing rows to contain the color-stripped observation.

File: pom.xml
Comment (jackah2) [inline pom.xml:7]:
```suggestion
	<version>2.5.0</version>
```
I'd say this is good to be a minor version bump

**Review by jackah2 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/whimc/observations/utils/sql/migration/schemas/Schema_4.java
Comment (samhum) [reply src/main/java/edu/whimc/observations/utils/sql/migration/schemas/Schema_4.java:19]: Still need to test my code for this want to do it locally before manipulating the actual observations table.

Comment (jackah2) [reply src/main/java/edu/whimc/observations/utils/sql/migration/schemas/Schema_4.java:36]: Create some observations using an older version on your test server and set the value inside `.schema_version` to `3` to have the migration run again.

I wouldn't put this plugin on the beta server until you're confident with the SQL changes on your local server.

File: src/main/java/edu/whimc/observations/models/Observation.java
Comment (jackah2) [inline src/main/java/edu/whimc/observations/models/Observation.java:71]: I was wrong, I think it would be better to just have this method be named `createPlayerObservation` with a JavaDoc explaining that it uses the configured expiration and calls the event; the name is pretty long.

Also, this is a nitpick but can you refactor the order of parameters and swap `observation` and `player` to be consistent with `loadObservation`

File: src/main/java/edu/whimc/observations/utils/sql/migration/schemas/Schema_4.java
Comment (jackah2) [inline src/main/java/edu/whimc/observations/utils/sql/migration/schemas/Schema_4.java:16]:
```suggestion
    /** Query to update existing observations with an uncolored version. */
```
Simple JavaDocs like these for variables can just be one-liners. However, I'd argue the variable names are self-documenting and don't warrant a JavaDoc

**Review by jackah2 — CHANGES_REQUESTED** [review]
Looks pretty good. I'd like to wait until the Checkstyle changes are merged in before merging any other PRs.

Can you confirm these changes work locally

**Comment by Geph** [comment]
More important to fix bugs than worry about checkstyles since Ateneo has another camp round soon. @samhum did you have a chance to test locally? I can compile it and stick it on the beta.

**Comment by Geph** [comment]
It's working locally except for not functioning as a quests module but that's the case for both of our plugins currently. @jackah2 and @samhum I'll wait on you two to sort out any other issues before merging in. The Checkstyles branch I have going resolves many of the issues this one is picking up.

**Comment by jackah2** [comment]
Resolves #35

**Pull request closed without merging**
